nncf/quantization/algorithms/smooth_quantize/algorithm.py

```diff
@@ -0,0 +1,262 @@
+                and point.target_point.type == TargetType.PRE_LAYER_OPERATION
+                and point.target_point.port_id == act_port
+            )
+
+        statistics_for_node = []
+        for tensor_collector in statistic_points.get_algo_statistics_for_node(node_name, filter_func, SmoothQuantize):
+            statistics_for_node.append(tensor_collector.get_statistics()[STATISTIC_BRANCH_KEY])
+        return statistics_for_node
+
+    def get_statistic_points(self, model: TModel) -> StatisticPointsContainer:
+        statistic_container = StatisticPointsContainer()
+
+        if self._alpha < 0:
+            nncf_logger.debug(
+                "Skipping statistics collection for SmoothQuantize algorithm because alfa parameter is negative."
+            )
+            return statistic_container
+
+        self._set_backend_entity(model)
+        nncf_graph = NNCFGraphFactory.create(model)
+
+        nodes_to_smooth_data = self._get_nodes_to_smooth_data(nncf_graph)
+
+        for node_data in nodes_to_smooth_data:
+            node_to_smooth = node_data["node_to_smooth"]
+            target_point = self._backend_entity.target_point(
+                TargetType.PRE_LAYER_OPERATION,
+                target_node_name=node_to_smooth.node_name,
+                port_id=node_data["input_act_port"],
+            )
+            input_reduction_shape = self._backend_entity.calculate_input_reduction_shape(
+                nncf_graph, node_to_smooth, node_data["input_act_port"]
+            )
+            stat_collector = self._backend_entity.get_abs_max_channel_collector(
+                self._subset_size, input_reduction_shape, self._inplace_statistics, STATISTIC_BRANCH_KEY
+            )
+            statistic_container.add_statistic_point(
+                StatisticPoint(
+                    target_point=target_point,
+                    tensor_collector=stat_collector,
+                    algorithm=SmoothQuantize,
+                )
+            )
+        return statistic_container
+
+    def _get_nodes_to_smooth_data(self, nncf_graph: NNCFGraph) -> List[Dict]:
+        """
+        Collects layers whose activations will be smoothed.
+
+        :param nncf_graph: NNCFGraph instance.
+        :return: List with the data for each layer.
+        """
+        nodes_with_weights = nncf_graph.get_nodes_by_metatypes(self._backend_entity.weighted_metatypes)
+        nodes_to_smooth_data = []
+
+        for node_with_weight in nodes_with_weights:
+            if not self._backend_entity.is_node_with_weights(node_with_weight):
+                continue
+
+            ports_map = self._backend_entity.get_input_ports_map(node_with_weight, nncf_graph)
+            weight_node = nncf_graph.get_input_edges(node_with_weight)[ports_map["weight"]].from_node
+
+            # Skipping shared weights
+            if len(nncf_graph.get_next_nodes(weight_node)) > 1:
+                continue
```

Comment:
This will not work for torch backend, I guess. Person who will be implementing this for the TORCH backend should come up with new common structure. I believe `layer_name` could be used for shared weights occasion as it done in MinMax algorithm

Reply:
Let's keep it for future